### PR TITLE
Fixed Model Branch

### PR DESCRIPTION
**Pull request by jalwin**
Please review and approve to be merged

### CODE REVIEW

File: PokemonApp/Pokemon Model/Pokemon.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import Foundation
+
+struct Pokemon: Codable {
+    let pokemonNumber: Int
+    let pokemonName: String
+    let sprites: PokemonSprite
+    let pokemonStats: [PokemonStats]
+    let pokemonTypes: [PokemonTypeContainer]
+    var gender: Bool?
+    var seen: Int? // 0 = never seen, 1 = seen, 2 = caught
```
Comment (prrenfrew) [inline PokemonApp/Pokemon Model/Pokemon.swift]: Probably should use an enum. You could give it a default value of never seen.

File: PokemonApp/Pokemon Model/PokemonSprite.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//  PokemonAppTests
+//
+//  Created by MAC Consultant on 6/14/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import Foundation
+
+struct PokemonSprite: Codable {
+    let back_default: String?
```
Comment (prrenfrew) [inline PokemonApp/Pokemon Model/PokemonSprite.swift]: Use lower camel case, like this: backDefault. These should be a bit more descriptive in the name as well.

File: PokemonApp/Pokemon Model/PokemonStatsModel/PokemonStat.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+//  Created by MAC Consultant on 6/14/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import Foundation
+
+struct PokemonStat: Codable {
+    let pokemonStatType: String
+    let url: String
+    enum CodingKeys: String, CodingKey {
```
Comment (prrenfrew) [inline PokemonApp/Pokemon Model/PokemonStatsModel/PokemonStat.swift]: There should be a blank line between the last property and the coding keys enum.

File: PokemonApp/Pokemon Model/PokemonStatsModel/PokemonStats.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//  PokemonAppTests
+//
+//  Created by MAC Consultant on 6/14/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import Foundation
+
+struct PokemonStats: Codable {
+    let base_stat: Int
```
Comment (prrenfrew) [inline PokemonApp/Pokemon Model/PokemonStatsModel/PokemonStats.swift]: Use camel case.

File: PokemonApp/PokemonApp/ViewController.swift [hunk 5]
```diff
@@ -9,8 +9,15 @@
...
 class ViewController: UIViewController {
+    
+    let apiURL = "https://pokeapi.co/api/v2/pokemon/ditto/"
 
   override func viewDidLoad() {
+    
+    let networkController = NetworkController()
+    networkController.getData(url: apiURL) { (fetchedPokemon: Pokemon?, error) in
+        print(fetchedPokemon?.pokemonName)
+    }
```
Comment (prrenfrew) [inline PokemonApp/PokemonApp/ViewController.swift]: What is the purpose of these changes?

Comment (jalwin) [reply PokemonApp/PokemonApp/ViewController.swift]: think a line or something was moved around as I was showing jerry how to use this. I will delete that and resolved.

File: PokemonApp/Pokemon Model/Pokemon.swift (hunk 1 above)
Comment (jalwin) [reply PokemonApp/Pokemon Model/Pokemon.swift]: Resolved. New enum file was created with raw values and assigned default raw value.

File: PokemonApp/Pokemon Model/PokemonSprite.swift (hunk 2 above)
Comment (jalwin) [reply PokemonApp/Pokemon Model/PokemonSprite.swift]: Resolved. gave more descriptive name.

File: PokemonApp/Pokemon Model/Pokemon.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+//  PokemonApp
+//
+//  Created by Alwin Amoros on 6/16/19.
+//  Copyright © 2019 PaulRenfrew. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+final class Pokemon: NSManagedObject, Codable {
```
Comment (prrenfrew) [inline PokemonApp/Pokemon Model/Pokemon.swift]: This class is duplicated.

File: PokemonApp/PokemonApp/AppDelegate.swift [hunk 7]
```diff
@@ -7,36 +7,83 @@
...
 
-  func applicationWillTerminate(_ application: UIApplication) {
+    func applicationWillTerminate(_ application: UIApplication) {
     // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
-  }
+    }
+    
+    // MARK: - Core Data stack
+    
+    lazy var persistentContainer: NSPersistentContainer = {
```
Comment (prrenfrew) [inline PokemonApp/PokemonApp/AppDelegate.swift]: I don't mind you having this for code, but don't commit and push this change.
